frontend/src/api/Socket.ts

```diff
@@ -1,54 +1,92 @@
+const DELETE_USER_URL:string = '/api/v1/socket/delete-user';
 
-export const sendUser = (nickname:string) => {
-  if (connected) {
-    stompClient.send(SEND_USER_URL, {}, nickname || 'Anonymous');
+export const isValidNickname = (nickname: string) => nickname.length > 0
+  && !nickname.includes(' ') && nickname.length <= 16;
+
+/**
+ * Add the user by sending a message via socket.
+ * @returns error, if present
+*/
+export const addUser = (nickname:string): Error | undefined => {
+  if (connected && isValidNickname(nickname)) {
+    stompClient.send(ADD_USER_URL, {}, nickname);
+  } else if (!connected) {
+    return new Error('The socket is not connected.');
   } else {
-    console.error('You must be connected to a socket before sending user information.');
+    return new Error('The provided nickname is invalid.');
   }
 };
 
-export const connect = (endpoint:string, nickname:string) => {
-  if (!connected) {
-    // Connect to given endpoint, subscribe to future messages, and send user message.
-    const socket: WebSocket = new SockJS(endpoint);
-    stompClient = Stomp.over(socket);
-    stompClient.connect({}, () => {
-      stompClient.subscribe(SUBSCRIBE_URL, (user: Message) => {
-        const userObject:User = JSON.parse(user.body);
-        console.log(`Welcome ${userObject.nickname} to the page!`);
-      });
-      sendUser(nickname);
-    });
-    // Reassign connected variable.
-    connected = true;
+/**
+ * Delete the user by sending a message via socket.
+ * @returns error, if present
+*/
+export const deleteUser = (nickname:string): Error | undefined => {
+  if (connected && isValidNickname(nickname)) {
+    stompClient.send(DELETE_USER_URL, {}, nickname);
+  } else if (!connected) {
+    return new Error('The socket is not connected.');
   } else {
-    console.error('You are already connected to a socket.');
+    return new Error('The provided nickname is invalid.');
   }
 };
 
-export const disconnect = () => {
+/**
+ * Connect and subscribe the user via socket.
+ * @returns error, if present
+*/
+export const connect = (endpoint:string, nickname:string,
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>):
+  Promise<Error | undefined> => new Promise<Error | undefined>((resolve, reject) => {
+    if (!connected) {
+      // Connect to given endpoint, subscribe to future messages, and send user message.
+      const socket: WebSocket = new SockJS(endpoint);
+      stompClient = Stomp.over(socket);
+      stompClient.connect({}, () => {
+        stompClient.subscribe(SUBSCRIBE_URL, (users: Message) => {
+          const userObjects:User[] = JSON.parse(users.body);
+          setUsers(userObjects);
+        });
+        // Reassign connected variable.
+        connected = true;
+        addUser(nickname);
+        resolve();
+      }, () => {
+        reject(new Error('The socket failed to connect.'));
+      });
+    } else {
+      reject(new Error('The socket is already connected.'));
```

Comment:
Two notes:
- I no longer have "connect" or "disconnect" buttons -- those are from an old PR. The "Enter" button on the "Enter Nickname" page will trigger the `connect(...)` function, but there will be no button that triggers `disconnect(...)` -- this will be triggered when the user leaves the page or is booted from the game or whatnot.
- This error handling is used even to check for cases that currently don't seem possible. Part of the point of error handling is to check for unexpected cases -- for example, so if something goes wrong where the user connects but they aren't transferred the the "Waiting Room" page, they won't re-connect, which will let them know to refresh the page.